packages/ses/test/test-scope.js

```diff
@@ -0,0 +1,341 @@
+/* global globalThis */
+
+import test from 'ava';
+import { makeSafeEvaluator } from '../src/make-safe-evaluator.js';
+
+test('scope behavior - lookup behavior', t => {
+  t.plan(7);
+
+  const globalObject = { foo: {} };
+  const globalLexicals = { foobar: {} };
+
+  const { safeEvaluate: evaluate } = makeSafeEvaluator({
+    globalObject,
+    globalLexicals,
+  });
+
+  globalThis.bar = {};
+
+  t.is(evaluate('globalThis'), undefined);
+  t.is(evaluate('eval'), undefined);
+  t.is(evaluate('bar'), undefined);
+  t.throws(() => evaluate('dummy'), { instanceOf: ReferenceError });
+
+  t.is(evaluate('foo'), globalObject.foo);
+  t.is(evaluate('foobar'), globalLexicals.foobar);
+
+  // Known compromise in fidelity of the emulated script environment:
+  t.deepEqual(evaluate('arguments'), ['arguments']);
+
+  delete globalThis.bar;
```

Comment:
Please setup as a [`teardown`](https://github.com/avajs/ava/blob/main/docs/02-execution-context.md#tteardownfn) cleanup in case the test fails to avoid polluting other tests.